aiida/orm/autogroup.py

```diff
@@ -7,173 +7,286 @@
+        self._group_label_prefix = default_label_prefix
+        self._group_label = None  # Actual group label, set by `get_or_create_group`
+
+    @staticmethod
+    def validate(strings):
+        """Validate the list of strings passed to set_include and set_exclude."""
+        if strings is None:
+            return
+        valid_prefixes = set(['aiida.node', 'aiida.calculations', 'aiida.workflows', 'aiida.data'])
+        for string in strings:
+            pieces = string.split(':')
+            if len(pieces) != 2:
+                raise exceptions.ValidationError(
+                    "'{}' is not a valid include/exclude filter, must contain two parts split by a colon".
+                    format(string)
+                )
+            if pieces[0] not in valid_prefixes:
+                raise exceptions.ValidationError(
+                    "'{}' has an invalid prefix, must be among: {}".format(string, sorted(valid_prefixes))
+                )
+
+            # If a % is present, it can only be the last character
+            string_without_percent = pieces[1]
+            if string_without_percent.endswith('%'):
+                string_without_percent = string_without_percent[:-1]
+            if '%' in string_without_percent:
+                raise exceptions.ValidationError(
+                    "'{}' can only contain a '%' character, if any, at the end of the string".format(string)
+                )
 
     def get_exclude(self):
-        """Return the list of classes to exclude from autogrouping."""
-        try:
-            return self.exclude
-        except AttributeError:
-            return []
+        """Return the list of classes to exclude from autogrouping.
 
-    def get_exclude_with_subclasses(self):
-        """
-        Return the list of classes to exclude from autogrouping.
-        Will also exclude their derived subclasses
-        """
-        try:
-            return self.exclude_with_subclasses
-        except AttributeError:
-            return []
+        Returns ``None`` if no exclusion list has been set."""
+        return self._exclude
 
     def get_include(self):
-        """Return the list of classes to include in the autogrouping."""
-        try:
-            return self.include
-        except AttributeError:
-            return []
-
-    def get_include_with_subclasses(self):
         """Return the list of classes to include in the autogrouping.
-        Will also include their derived subclasses."""
-        try:
-            return self.include_with_subclasses
-        except AttributeError:
-            return []
+
+        Returns ``None`` if no exclusion list has been set."""
+        return self._include
+
+    def get_group_label_prefix(self):
+        """Get the prefix of the label of the group.
+        If no group label prefix was set, it will set a default one by itself."""
+        return self._group_label_prefix
 
     def get_group_name(self):
```

Comment:
Since we are anyway breaking backwards compatibility, should we rename this to `get_group_name`?

Reply:
I kept it for backward compatibility (the function is deprecated). As you noticed, I ended up removing other stuff that is not really deprecated, but this was internal logic anyway (and not working until now) so I don't think it's a problem.